server/app/forms/admin/ProgramStatusesEditForm.java

```diff
@@ -0,0 +1,77 @@
+package forms.admin;
+
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import play.data.validation.Constraints;
+import play.data.validation.Constraints.Validatable;
+import play.data.validation.Constraints.Validate;
+import play.data.validation.ValidationError;
+
+/** Form for creating / editing program statuses. */
+@Validate
+public final class ProgramStatusesEditForm implements Validatable<List<ValidationError>> {
+  public static final String ORIGINAL_STATUS_TEXT_FORM_NAME = "originalStatusText";
+  public static final String STATUS_TEXT_FORM_NAME = "statusText";
+  public static final String EMAIL_BODY_FORM_NAME = "emailBody";
+
+  public static final ImmutableList<String> FIELD_NAMES =
+      ImmutableList.of(ORIGINAL_STATUS_TEXT_FORM_NAME, STATUS_TEXT_FORM_NAME, EMAIL_BODY_FORM_NAME);
+
+  private String originalStatusText;
+
+  @Constraints.Required(message = "This field is required.")
+  private String statusText;
+
+  private String emailBody;
+
+  public ProgramStatusesEditForm() {
+    this.originalStatusText = "";
+    this.statusText = "";
+    this.emailBody = "";
+  }
+
+  public String getOriginalStatusText() {
+    return originalStatusText != null ? originalStatusText : "";
```

Comment:
We're using null a lot here when I don't think it is necessary; eg required by the framework or a dependency.

We should use Optional unless we can't, EG for non required DB values etc.

Reply:
Empirically, the framework returns `null` if the form parameter is missing entirely from the request body ([docs](https://www.playframework.com/documentation/2.8.x/JavaForms)). We aren't in control of how this data is populated since we're using the `form(...)` helper and binding from the request.